app/src/main/res/layout/recorder_fragment.xml

```diff
@@ -1,57 +1,88 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
     tools:context=".ui.activities.MainActivity">
 
     <ImageView
         android:id="@+id/record_header_image"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintEnd_toEndOf="parent"
         android:layout_marginTop="64dp"
-        />
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
     <TextView
         android:id="@+id/record_filename"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
-        android:text="Press the mic button \n to start recording"
-        android:textSize="18sp"
-        android:lineHeight="32dp"
-        android:textAlignment="center"
         android:layout_marginStart="32dp"
+        android:layout_marginTop="64dp"
         android:layout_marginEnd="32dp"
-        app:layout_constraintTop_toBottomOf="@id/record_header_image"
-        app:layout_constraintStart_toStartOf="parent"
+        android:lineHeight="32dp"
+        android:text="Press the mic button \n to start recording"
+        android:textAlignment="center"
+        android:textSize="18sp"
         app:layout_constraintEnd_toEndOf="parent"
-        android:layout_marginTop="64dp"
-        />
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/record_header_image" />
+
+    <RelativeLayout
```

Comment:
no need to use RelativeLayout, the layout built using ConstraintLayout all you neet is to constraint the VisulaizerView to its place, if you find it difficult let me know

Reply:
also, try to make the background of the Visualizer to transparent / white

Reply:
👍